File: src/main/java/com/SWEProject/Entities/Product.java
package com.SWEProject.Entities;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

@Entity
public class Product {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Integer id;
	private String name;
	private float lower;
	private float upper;
	private String category;
	private String type;   // online or offline
	private boolean onsystem;
	@ManyToOne
	private Brand brand;
	
	@OneToMany(targetEntity = StoreProducts.class, cascade = CascadeType.ALL)
	private Set<StoreProducts> stores;
	
	public Product() {
		this.name = "";
		this.lower = -1;
		this.upper = -1;
		this.category = "";
		this.type = "";   // online or offline
		this.brand = null;
		this.stores = new HashSet<StoreProducts>();
		this.onsystem = false;
	}
	
	public Product(String name, float lower, float upper, String category, String type, Brand brand,
			Set<StoreProducts> stores, boolean onsystem) {
		super();
		this.name = name;
		this.lower = lower;
		this.upper = upper;
		this.category = category;
		this.type = type;
		this.brand = brand;
		this.stores = stores;
		this.onsystem = onsystem;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public float getLower() {
		return lower;
	}

	public void setLower(float lower) {
		this.lower = lower;
	}

	public float getUpper() {
		return upper;
	}

	public void setUpper(float upper) {
		this.upper = upper;
	}

	public String getCategory() {
		return category;
	}

	public void setCategory(String category) {
		this.category = category;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public Brand getBrand() {
		return brand;
	}

	public void setBrand(Brand brand) {
		this.brand = brand;
	}

	public Set<StoreProducts> getStores() {
		return stores;
	}

	public void setStores(Set<StoreProducts> stores) {
		this.stores = stores;
	}

	public boolean isOnsystem() {
		return onsystem;
	}

	public void setOnsystem(boolean onsystem) {
		this.onsystem = onsystem;
	}

}

File: front-end/src/app/models/product.ts
import { Brand } from './Brand';
import { StoreProduct } from "./StoreProduct";

export class Product{
    private id: number;

    /**
     * Getter $upper
     * @return {number}
     */
	public get $upper(): number {
		return this.upper;
	}

    /**
     * Setter $upper
     * @param {number} value
     */
	public set $upper(value: number) {
		this.upper = value;
	}

    /**
     * Getter $category
     * @return {string}
     */
	public get $category(): string {
		return this.category;
	}

    /**
     * Setter $category
     * @param {string} value
     */
	public set $category(value: string) {
		this.category = value;
	}

    /**
     * Getter $type
     * @return {string}
     */
	public get $type(): string {
		return this.type;
	}

    /**
     * Setter $type
     * @param {string} value
     */
	public set $type(value: string) {
		this.type = value;
	}

    /**
     * Getter $onsystem
     * @return {boolean}
     */
	public get $onsystem(): boolean {
		return this.onsystem;
	}

    /**
     * Setter $onsystem
     * @param {boolean} value
     */
	public set $onsystem(value: boolean) {
		this.onsystem = value;
	}

    /**
     * Getter $brand
     * @return {Brand}
     */
	public get $brand(): Brand {
		return this.brand;
	}

    /**
     * Setter $brand
     * @param {Brand} value
     */
	public set $brand(value: Brand) {
		this.brand = value;
	}

    /**
     * Getter $stores
     * @return {StoreProduct[]}
     */
	public get $stores(): StoreProduct[] {
		return this.stores;
	}

    /**
     * Setter $stores
     * @param {StoreProduct[]} value
     */
	public set $stores(value: StoreProduct[]) {
		this.stores = value;
	}

    /**
     * Getter $name
     * @return {string}
     */
	public get $name(): string {
		return this.name;
	}

    /**
     * Setter $name
     * @param {string} value
     */
	public set $name(value: string) {
		this.name = value;
	}

    /**
     * Getter $id
     * @return {number}
     */
	public get $id(): number {
		return this.id;
	}

    /**
     * Setter $id
     * @param {number} value
     */
	public set $id(value: number) {
		this.id = value;
	}
	private name: string;

    /**
     * Getter $lower
     * @return {number}
     */
	public get $lower(): number {
		return this.lower;
	}

    /**
     * Setter $lower
     * @param {number} value
     */
	public set $lower(value: number) {
		this.lower = value;
	}
	private lower: number;
	private upper: number;
	private category: string;
	private type: string;   // online or offline
	private onsystem: boolean;
	private brand: Brand;
    private stores: StoreProduct[];
    
    constructor(){

    }
    setData(Data): void{
        this.id = Data.id;
        this.name = Data.name;
        this.lower = Data.lower;
        this.upper = Data.upper;
        this.category = Data.category;
        this.type = Data.type;
        this.onsystem = Data.onsystem;
        this.brand = Data.brand;
        this.stores = Data.stores;
    }
    
}